src/CommunityToolkit.Maui.Core/Essentials/FileFolderDialog.tizen.cs

```diff
@@ -0,0 +1,398 @@
+			HeightSpecification = LayoutParamPolicies.WrapContent
+		};
+		content.Add(hlayout);
+
+		var cancelButton = new Button
+		{
+			Focusable = true,
+			Text = "Cancel",
+			TextColor = TColor.Black,
+			BackgroundColor = TColor.Transparent.ToNative(),
+		};
+		cancelButton.TextLabel.PixelSize = 15d.ToPixel();
+		cancelButton.SizeWidth = cancelButton.TextLabel.NaturalSize.Width + 15d.ToPixel() * 2;
+		cancelButton.Clicked += (s, e) => SendCancel();
+		hlayout.Add(cancelButton);
+
+		var okButton = new Button
+		{
+			Focusable = true,
+			Text = isFileSelectionMode ? "Save" : "OK",
+			TextColor = TColor.Black,
+			BackgroundColor = TColor.Transparent.ToNative(),
+		};
+		okButton.TextLabel.PixelSize = 15d.ToPixel();
+		okButton.SizeWidth = okButton.TextLabel.NaturalSize.Width + 15d.ToPixel() * 2;
+		okButton.Clicked += (s, e) =>
+		{
+			if (isFileSelectionMode)
+			{
+				if (IsDirectory(selectedPath))
+				{
+					selectedFileName = fileNameEntry is null ? selectedFileName : fileNameEntry.Text;
+					SendSubmit(selectedPath + selectedFileName);
+				}
+				else
+				{
+					SendSubmit(selectedPath);
+				}
+			}
+			else
+			{
+				SendSubmit(selectedPath);
+			}
+		};
+		hlayout.Add(okButton);
+
+		Relayout += (s, e) =>
+		{
+			var isHorizontal = Window.Instance.WindowSize.Width > Window.Instance.WindowSize.Height;
+			content.SizeWidth = Window.Instance.WindowSize.Width * (isHorizontal ? 0.5f : 0.8f);
+		};
+
+		return content;
+	}
+
+	static bool CreateSubDirectory(string newDirectory)
+	{
+		Directory.CreateDirectory(newDirectory);
+		return Directory.Exists(newDirectory);
+	}
+
+	void ProcessSelect(string selectedItem)
+	{
+		if (selectedItem.Equals(previousDirectorySymbol, StringComparison.Ordinal))
+		{
+			if (selectedPath.Equals(initialFileFolderPath, StringComparison.Ordinal)
+				|| (selectedPath + slashSymbol).Equals(initialFileFolderPath, StringComparison.Ordinal))
+			{
+				return;
+			}
+			selectedPath = Path.GetDirectoryName(selectedPath) ?? initialFileFolderPath;
+			selectedPath = selectedPath[..selectedPath.LastIndexOf(slashSymbol, StringComparison.Ordinal)];
+		}
+		else
+		{
+			if (!IsDirectory(selectedPath))
+			{
+				selectedPath = Path.GetDirectoryName(selectedPath) ?? selectedPath[..selectedPath.LastIndexOf(slashSymbol, StringComparison.Ordinal)];
+			}
+			selectedPath = selectedPath.EndsWith(slashSymbol) ? selectedPath : selectedPath + slashSymbol;
+			selectedPath += selectedItem;
+		}
+
+		if (IsDirectory(selectedPath))
+		{
+			UpdateDirectoryScrollView(selectedPath);
+		}
+		else if (isFileSelectionMode)
+		{
+			fileNameEntry!.Text = selectedItem;
+		}
+	}
+
+	void UpdateDirectoryScrollView(string path)
+	{
+		if (directoryScrollView is null)
+		{
+			return;
+		}
+
+		foreach(var view in directoyViews)
+		{
+			directoryScrollView.ContentContainer.Remove(view);
+		}
+		directoyViews.Clear();
+
+		var listItems = GetDirectories(path);
+
+		if (listItems is not null)
```

Comment:
can we invert this if, to better readability

Reply:
sure